src/scriptworker/cot/verify.py

```diff
@@ -1281,6 +1281,29 @@ async def populate_jsone_context(chain, parent_link, decision_link, tasks_for):
 
 
 # get_in_tree_template {{{1
+def build_taskcluster_yml_url(link):
+    """Build the url to the repo root ``.taskcluster.yml`` .
+
+    Args:
+        link (LinkOfTrust): the parent link to get the source url from.
+
+    Returns:
+        string: the ``.taskcluster.yml`` url
+
+    """
+    source_env_prefix = link.context.config["source_env_prefix"]
+    repo_url = get_repo(link.task, source_env_prefix)
+    repo_url = repo_url.replace("git@github.com:", "ssh://github.com/", 1)
+    revision = get_revision(link.task, source_env_prefix)
+    repo_parts = urlparse(repo_url)
+    if repo_parts.netloc == "github.com":
+        user, repo_name = extract_github_repo_owner_and_name(repo_url)
+        path_prefix = f"/{user}/{repo_name}/raw"
+    else:
+        path_prefix = f"{repo_parts.path}/raw-file"
+    return f"{repo_parts.scheme}://{repo_parts.netloc}{path_prefix}/{revision}/.taskcluster.yml"
```

Comment:
I think this would be better if the two branches didn't try to share code. While both hg.mozilla.org and github happen to have similiar URL structures[1], I don't think trying to reduce duplication makes things clearer. If one of these needed to change for any reason[2], it is unlikely that the other branch will need to change, which will make maintenance harder rather than easier.

---

[1] probably because it is a fairly natural URL strucutre
[2] perhaps so that github always uses raw.githubusercontent.com